test/test_distributions.py

```diff
@@ -4025,6 +4025,76 @@ def f(x):
             self.assertEqual(f(x), traced_f(x))
 
 
+class TestFunctors(TestCase):
+    def test_cat_transform(self):
```

Comment:
It would be good to add one more `test_cat_transform_lengths` with an example with nonuniform lengths. It's nice to test both code paths: uniform lengths (with lengths not provided) and nonuniform lengths.

Reply:
Sure, will do!